### ISSUE TITLE
get0 linter

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1970]
A pattern we see occasionally looks like:

```r
if (exists("key", ...)) {
  value <- get("key", ...)
} else {
  value <- default()
}
```

With some minor variations thereof.

`base::get0()` is a more concise and efficient alternative in this case:

```r
value <- get0("key", ..., ifnotfound = default())
```

The problem comes from getting the logic here right... it's easy to see a `get0()` replacement manually, but tough to design a rule that will catch enough cases for it to be worth it.

In our code, I did a simple pass like this:

```
//IF[
  following-sibling::expr[1]/expr/SYMBOL_FUNCTION_CALL[text() = 'exists']
  and following-sibling::expr[2]//SYMBOL_FUNCTION_CALL[text() = 'get']
  and following-sibling::ELSE
]
```

There were few enough hits that I could filter through the false positives manually. But if we want to productionize this we should try to exclude more false positives up front.
